components/scripts/lib/build-scan-online.sh

```diff
@@ -7,7 +7,7 @@ readonly FETCH_BUILD_SCAN_DATA_JAR="${LIB_DIR}/export-api-clients/fetch-build-sc
 # Enterprise API.
 process_build_scan_data_online() {
   read_build_scan_metadata
-  fetch_and_read_build_scan_data build_cache_metrics_only
+  fetch_and_read_build_scans build_cache_metrics_only "${build_scan_urls[@]}"
```

Comment:
I keep forgetting why we need this `build_cache_metrics_only` flag and logic. Is it an over-optimization?

Reply:
I had a similar thought while refactoring. It all boils down to this `if` statement in the parse logic: 

https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/ecbb2ae2a328eb22c57bdf41724781b3a9f0e786/components/scripts/lib/build-scan-parse.sh#L64-L74

For CI/CI experiments the value is set to `all_data` because we don't have any information about the builds (e.g. outcome, requested tasks, etc.)

For CI/Local the **_first_** request to fetch the CI build is also `all_data` for a similar reason.

In all other cases it is set to `build_cache_metrics_only`. 

My thought was why not just parse all data all the time? We will override some values we already knew and obtained from the environment (e.g. requested tasks and build outcome). I think the only values the Build Scan may not have are the Git details in the scenario where CCUD is not applied. It would be nice to not have to pass the `build_cache_metrics_only` around.

Reply:
`My thought was why not just parse all data all the time? We will override some values we already knew and obtained from the environment (e.g. requested tasks and build outcome). I think the only values the Build Scan may not have are the Git details in the scenario where CCUD is not applied. It would be nice to not have to pass build_cache_metrics_only around to all these functions.`

Yes, we are fully aligned in our thoughts here. Can you add it to our doc, please, and then resolve the thread here?

Reply:
Done.